pkg/graveler/graveler.go

```diff
@@ -892,7 +895,71 @@ func (g *KVGraveler) WriteMetaRangeByIterator(ctx context.Context, repositoryID
 	return g.CommittedManager.WriteMetaRangeByIterator(ctx, repo.StorageNamespace, it, nil)
 }
 
+func (g *KVGraveler) deleteRepositoryBranches(ctx context.Context, repositoryID RepositoryID) error {
+	itr, err := g.ListBranches(ctx, repositoryID)
+	if err != nil {
+		return err
+	}
+	defer itr.Close()
+	var wg multierror.Group
+	for itr.Next() {
+		b := itr.Value()
+		wg.Go(func() error {
+			return g.RefManager.DeleteBranch(ctx, repositoryID, b.BranchID)
+		})
+	}
+	return wg.Wait().ErrorOrNil()
+}
+
+func (g *KVGraveler) deleteRepositoryTags(ctx context.Context, repositoryID RepositoryID) error {
+	itr, err := g.ListTags(ctx, repositoryID)
+	if err != nil {
+		return err
+	}
+	defer itr.Close()
+	var wg multierror.Group
+	for itr.Next() {
+		tag := itr.Value()
+		wg.Go(func() error {
+			return g.DeleteTag(ctx, repositoryID, tag.TagID)
+		})
+	}
+	return wg.Wait().ErrorOrNil()
+}
+
+func (g *KVGraveler) deleteRepositoryCommits(ctx context.Context, repositoryID RepositoryID) error {
+	itr, err := g.RefManager.ListCommits(ctx, repositoryID)
+	if err != nil {
+		return err
+	}
+	defer itr.Close()
+	var wg multierror.Group
+	for itr.Next() {
+		commit := itr.Value()
+		wg.Go(func() error {
+			return g.RefManager.RemoveCommit(ctx, repositoryID, commit.CommitID)
+		})
+	}
+	return wg.Wait().ErrorOrNil()
+}
```

Comment:
I'm unsure that creating (#Tags + #Commits + #Branches) goroutines is a good idea. You may be needing workers to make sure you're not creating a 1M routines for large repositories.

Reply:
@itaiad200  Do you think it's better to just serialize the calls for now? This is just temporary solution, which obviously we will not release KV with

Reply:
For a temporary solution it's fine for now.